graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/builtins/commonjs/CommonJSRequireBuiltin.java

```diff
@@ -169,6 +171,21 @@ protected static Object fallback(@SuppressWarnings("unused") Object function, Ob
     @TruffleBoundary
     private Object requireImpl(String moduleIdentifier, TruffleFile entryPath, JSRealm realm) {
         log("required module '", moduleIdentifier, "' from path ", entryPath);
+        // 1.1 (Non-spec): If a module resolver hook has been installed, give it a chance to resolve the module, but
+        // only if `handler` mode is enabled for JS module resolution.
+        if (realm.getContextOptions().getModuleLoaderFactoryMode().equals(HANDLER)) {
```

Comment:
should be `HANDLER.equals(realm.getContextOptions().getModuleLoaderFactoryMode())` to avoid npe